### PR TITLE
Add git sync v4

### PR DESCRIPTION
**Pull request by jaytmiller**
Adds a 4th independent git-sync rewrite which has the same general strategy of v2 and v3 where clone directories are kept r/w for notebook execution but files/notebooks are locked down as readonly to discourage direct use for development.  Major aspects of the rewrite are to switch to Python,  add unit tests,  and to attempt to differentiate between user-owned/modified files and git-owned files to avoid altering user files,  even file perms.  Also attempts to directly communicate with users using a SYNC-INSTRUCTIONS.md file to explain intended usage,  and provides an opt-out mechanism for e.g. TIKE users who don't care about updates.  Also stops using default repo names and moves all clones to a references/ directory.   Affects all missions,  all post-start-hooks updated with the intent that most failures cannot block logins,  with EFS quota violations being the one failure I recall that can block logins.   Also adds a gs4.fail.log file to the repo directory,  assuming the repo dir exists but the sync failed hard for some reason;  deleted for successful syncs.

### CODE REVIEW

File: deployments/common/common-scripts/SYNC-INSTRUCTIONS.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+## Automatic Updates
+
+This reference directory is sync'ed with its source repository each time you
+log in so that you automatically receive the latest materials the authors have
+committed to the source repository. If however you modify any files or
+directories in place, upon sync your changes will be backed up and the original
+file locations reverted and updated normally.  Likewise, if you save a file to
+a new name, and the source repository coincidentally adds the same file name
+later with different contents, during the sync your version of the file will be
+backed up to a different name.  To the extent possible, after evey sync the
```
Comment (msanchezst) [inline deployments/common/common-scripts/SYNC-INSTRUCTIONS.md:67]: minor typo (every sync)

File: deployments/common/pkgs/git-sync-v4/design.txt
Comment (msanchezst) [inline deployments/common/pkgs/git-sync-v4/design.txt:1]: are we including this file?

File: deployments/jwebbinar/environments/post-start-hook [hunk 2]
```diff
@@ -1,51 +1,74 @@
...
 
-# XXXXXX PAY ATTENTION XXXXXXXXXXXXXX
-# Notebooks are configured for rush jwebbinar-28 downloaded from jaytmiller fork.
-# This definitely has to change for the next webbinar.
-
-cd $HOME
-/opt/common-scripts/git-sync https://github.com/spacetelescope/jwebbinar_prep  jwebbinar_29  /home/jovyan/jwebbinar_prep
-chmod -R +w /home/jovyan/jwebbinar_prep
-
+# .............................................................................................
```
Comment (msanchezst) [inline deployments/jwebbinar/environments/post-start-hook:51]: We can probably delete these old approaches to remove some clutter. thoughts?

**Review by msanchezst — APPROVED** [review]
Wow, this was a lot of work and so many moving pieces. It looks fantastic to me. Thanks for putting all this together and all the planning that went into it.

File: deployments/common/common-scripts/SYNC-INSTRUCTIONS.md (hunk 1 above)
Comment (jaytmiller) [reply deployments/common/common-scripts/SYNC-INSTRUCTIONS.md:67]: Got it

File: deployments/common/pkgs/git-sync-v4/design.txt
Comment (jaytmiller) [reply deployments/common/pkgs/git-sync-v4/design.txt:1]: Removed

File: deployments/jwebbinar/environments/post-start-hook (hunk 2 above)
Comment (jaytmiller) [reply deployments/jwebbinar/environments/post-start-hook:51]: I'll try to clean that up some,  one commit coming for this and preceding comments.

Comment (jaytmiller) [reply deployments/jwebbinar/environments/post-start-hook]: Updates committed.

**Review by msanchezst — APPROVED** [review]
Looks great!

**Pull request merged**